### PR TITLE
animate hyena saving log on project page

### PR DESCRIPTION
**Pull request by sosuke-k**
## Why
- I'd like to look back over the `hyena save` history graphically and restore the past state.
### Animation
- [Jarallax](http://www.jarallax.com/)
### Data
- Hyena History API returns following data:
  - `-1` means this line is still not added or already deleted.
  - `0` means the current line number and the last line number of this line are same.
    - e.g. `2,0,2,2,2` -> `2,2,2,2,2`、`3,0,-1,0,0` -> `3,3,-1,-1,-1`
  - other is the line number of this line.
    [Gyazo](http://gyazo.com/3f916aa4cb75133247a4c593e695882f)

### CODE REVIEW

**Comment by sosuke-k** [comment]
## Wanted

* スクロールバーにコミット毎の節目
    * イメージ
    

* 復元ボタン（GUIで表示しているコミットがされた状態を復元する）

**Comment by sosuke-k** [comment]
@rooa 
I added the following new files:
- root/
  - static/
    - elements/
      - project-history-jarallax.html
    - javascripts/
      - jarallax-0.2.4b.js
      - jquery-ui.js
      - jquery.js
      - jquery.mousewheel.min.js
    - stylesheets/
      - reset.css
  - templates/
    - project.html
- util/
  - git/
    - repository.go (before git.go)
    - struct.go
    - struct_test.go
    - struct_test_data.go
    - history.go
    - history_test.go
    - test_repository/
  - re/re.go

but, please skip javascripts folder and stylesheets folder
and you still don't have to read the files related to testing (e.g. struct_test.go)
because I don't really recognize testing...TT

**Comment by rooa** [comment]
@sosuke-k thanks, I'll look up the codes but it seem that it takes time for me to finish reading. I'll check it by tmr.

File: util/git/struct_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			So(commit.Diffs[0].A.LineNumbers[0], ShouldEqual, 3)
+		})
+	})
+}
+
+func TestDiffInfo(t *testing.T) {
+
+	Convey("contains with n", t, func() {
+
+		Convey("How can i write this test", nil)
```
Comment (rooa) [inline util/git/struct_test.go:47]: :innocent:

Comment (sosuke-k) [inline util/git/struct_test.go:47]: 汗

**Comment by rooa** [comment]
Can `2,0,2,2,2` happen? In this case it should be `2,0,0,0,0` if my understanding is correct.

**Comment by sosuke-k** [comment]
it should be `2,0,2,2,2` because

**Comment by sosuke-k** [comment]
In this case there are 5 commits
and chrome.json was no changed in 2nd commit.
[Gyazo](http://gyazo.com/c8b652e5e9189222acec8cb2c024a33a)

**Comment by sosuke-k** [comment]
https://github.com/sosuke-k/hyena/pull/76/files#diff-0e7214ef20fcc722f1c5a43fe44d2afdR139
In this line, `LineNumberSequence.push({line number})` ,
but no diff can not run this line.
so
https://github.com/sosuke-k/hyena/pull/76/files#diff-0e7214ef20fcc722f1c5a43fe44d2afdR164
In this line , I push 0 to regulate LineNumberSequence's length.

**Comment by rooa** [comment]
So it's a dummy 0 when there's no change in the commit of a file.
I see

**Comment by rooa** [comment]
@sosuke-k lgtm

**Comment by sosuke-k** [comment]
@rooa I'll merge it.

**Pull request merged**